sympy/sets/fancysets.py

```diff
@@ -910,14 +909,29 @@ def _boundary(self):
 
     def as_relational(self, x):
         """Rewrite a Range in terms of equalities and logic operators. """
-        from sympy.functions.elementary.integers import floor
         if self.size == 1:
             return Eq(x, self[0])
+        elif self.size == 0:
+            return S.false
         else:
-            return And(
-                Eq(x, floor(x)),
-                x >= self.inf if self.inf in self else x > self.inf,
-                x <= self.sup if self.sup in self else x < self.sup)
+            from sympy.core.mod import Mod
+            from sympy.sets.contains import Contains
+            cond = None
+            if self.start.is_infinite:
+                if self.stop.is_infinite:
+                    cond = S.true
+                else:
+                    a = self.reversed.start
+            elif self.start == self.stop:
+                cond = S.false  # null range
+            else:
+                a = self.start
+            step = abs(self.step)
+            cond = Eq(Mod(x, step), a % step) if cond is None else cond
+            return And(Contains(x, S.Integers),
```

Comment:
The Contains is Set notation and this is the `as_relational` method. This addition is not necessary since only integers will be permitted in the Relationals.

Reply:
The following could be opened as a new issue or solved in the PR, too:
```python
>>> Integers.as_relational(x)  # and similar for Naturals and Naturals0
(-oo < x) & (x < oo) & Eq(floor(x), x) -> (-oo < x) & (x < oo) & Eq(Mod(x, 1), 0)
```

Reply:
> The following could be opened as a new issue or solved in the PR, too:

I checked this issue, Its not only `Integers.as_relational(x)`  but `Naturals.as_relational(x)` also need modifications.
But even the present code is giving different errors. Maybe I am doing something wrong,please check the following code --
I tried to check how it works , And I am getting these errors,

```
>>>from sympy.sets.fancysets import Integers, Naturals
>>>from sympy.abc import x,y
>>>Integers.as_relational(x)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-15-f65c72d2dd7f> in <module>
----> 1 Integers.as_relational(x)

TypeError: as_relational() missing 1 required positional argument: 'x'
```
But if I add an extra argument :
```

>>>Integers.as_relational(x,y)
⌊𝑦⌋=𝑦∧−∞<𝑦∧𝑦<∞
```

It works fine. 
But the `Naturals.as_relational(x)` does not work:

```
>>>Naturals.as_relational(x)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-20-cc0df9457647> in <module>
      1 from sympy.sets.fancysets import Integers, Naturals
      2 from sympy.abc import y
----> 3 Naturals.as_relational(x)

TypeError: as_relational() missing 1 required positional argument: 'x'

>>>Naturals.as_relational(x,y)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-18-27af4f8444d5> in <module>
----> 1 Naturals.as_relational(x,y)

c:\users\user\appdata\local\programs\python\python38-32\lib\site-packages\sympy\sets\fancysets.py in as_relational(self, x)
    130     def as_relational(self, x):
    131         from sympy.functions.elementary.integers import floor
--> 132         return And(Eq(floor(x), x), x >= self.inf, x < oo)
    133 
    134 

AttributeError: 'Symbol' object has no attribute 'inf'
```

 I want to open an new issue and work on it. Can I do that?

Reply:
Sure. But I can't reproduce your issue:

```python
>>> from sympy import Integers
>>> Integers.as_relational(x)
(-oo < x) & (x < oo) & Eq(floor(x), x)
```

I can also commit this PR after you remove the Contains argument. @asmeurer ?

Reply:
> I can also commit this PR after you remove the Contains argument. @asmeurer ?

I have removed the `Contains` argument from code.

Reply:
> The following could be opened as a new issue or solved in the PR, too:
> 
> ```python
> >>> Integers.as_relational(x)  # and similar for Naturals and Naturals0
> (-oo < x) & (x < oo) & Eq(floor(x), x) -> (-oo < x) & (x < oo) & Eq(Mod(x, 1), 0)
> ```

And I am referencing this in a new issue to work on.

Reply:
> The following could be opened as a new issue or solved in the PR, too:
> 
> ```python
> >>> Integers.as_relational(x)  # and similar for Naturals and Naturals0
> (-oo < x) & (x < oo) & Eq(floor(x), x) -> (-oo < x) & (x < oo) & Eq(Mod(x, 1), 0)
> ```

Sir there is one more thing, that `as_relational` rewrite an interval in terms of inequalities and logic operators. So if `Integers.as_relational(x)` is logically correct with `floor` then isn't `(-oo < x) & (x < oo) & Eq(floor(x), x) & Eq(Mod(x, 1), 0)
`  will be more accurate output.

Reply:
> be more accurate output.

It's more accurate if it gives a different result for some x than the other expression (and that different result is right). For what value of `x` do the two versions of the relational give different answers/results?